sharness/lib/test-lib.sh

```diff
@@ -27,7 +27,7 @@ test_ipfs_init() {
         echo "Docker not found"
         exit 1
     fi
-    if docker ps -a --format '{{.Names}}' | egrep -q '^ipfs$'; then
```

Comment:
Why did you drop the "-a"?  I didn't research the implications of adding it fully, but I added this in the latest fix-sharness PR because docker ps without "-a" couldn't find my running ipfs container and kept causing a tricky crash when running sharness several times in a row.

Reply:
Yeah I thought it was no big deal then. The -a means  "Show all containers (default shows just running)". But debugging myself such error I was getting the "ipfs container already running" message when the ipfs container was not in running state (there was a stuck one in Created state I think, probably same issue you had). which was not accurate.

I'm going to open an issue about hardening this shell script a bit. With our without -a this should attempt to kill/remove existing containers with that name or log more info.